Sources/TerminalAPIKit/Crypto/AES.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  AES.swift
+//  TerminalAPIKit
+//
+//  Copyright (c) 2022 Adyen N.V.
+//
+
+import Foundation
+import CommonCrypto
+
+/// Performs an AES encryption or decryption in CBC mode using PKCS#7 padding.
+///
+/// - Parameters:
+///   - operation: The operation to perform.
+///   - input: The data to encrypt or decrypt.
+///   - key: The key material to use.
+///   - initializationVector: The initialization vector to use. Must be equal in length to an AES128 block size.
+/// - Returns: The result of the operation.
+/// - Throws: ``AESError/operationError(status:)`` if the operation was not successful.
+/// - Throws: ``AESError/invalidKeySize`` if the provided `key` size is not 256 bits.
+/// - Throws: ``AESError/invalidInitializationVectorSize`` if the provided `initializationVector` size is not 128 bits.
+public func aes(
+    _ operation: AESOperation,
+    input: Data,
+    key: Data,
+    initializationVector: Data
+) throws -> Data {
+    
+    guard key.count == kCCKeySizeAES256 else {
+        throw AESError.invalidKeySize
+    }
+    
+    guard initializationVector.count == kCCBlockSizeAES128 else {
+        throw AESError.invalidInitializationVectorSize
+    }
+    
+    var outputLength = operation.outputLength(forInputLength: input.count)
+    var output = Data(count: outputLength)
+    
+    let status = output.withUnsafeMutableBytes { outputPointer in
+        input.withUnsafeBytes { inputPointer in
+            key.withUnsafeBytes { keyPointer in
+                initializationVector.withUnsafeBytes { ivPointer in
+                    CCCrypt(operation.ccOperation,
+                            CCAlgorithm(kCCAlgorithmAES),
+                            CCOptions(kCCOptionPKCS7Padding),
+                            keyPointer.baseAddress,
+                            key.count,
+                            ivPointer.baseAddress,
+                            inputPointer.baseAddress,
+                            input.count,
+                            outputPointer.baseAddress,
+                            outputLength,
+                            &outputLength)
+                }
+            }
+        }
+    }
+    
+    guard status == kCCSuccess else {
+        throw AESError.operationError(status: status)
+    }
+    
+    return output.prefix(upTo: outputLength)
+}
+
```

Comment:
```suggestion
    guard initializationVector.count == kCCBlockSizeAES128 else {
        throw AESError.invalidInitializationVectorSize
    }
    
    var outputLength = operation.outputLength(forInputLength: input.count)
    var outputBytes = [UInt8]()
    outputBytes.reserveCapacity(outputLength)
    
    let status = CCCrypt(operation.ccOperation,
                         CCAlgorithm(kCCAlgorithmAES),
                         CCOptions(kCCOptionPKCS7Padding),
                         key.bytes,
                         key.count,
                         initializationVector.bytes,
                         input.bytes,
                         input.count,
                         &outputBytes,
                         outputLength,
                         &outputLength)
    
    guard status == kCCSuccess else {
        throw AESError.operationError(status: status)
    }
    
    return Data(outputBytes.prefix(upTo: outputLength))
}

extension Data {
    var bytes: [UInt8] {
        return [UInt8](self)
    }
}
```

Reply:
unfortunately this breaks the tests (at least on my machine). have you tried running it?

Reply:
fixed it!, tests succeed on my laptop

Reply:
thanks! this works now 👍